ostd/src/user.rs

```diff
@@ -108,14 +109,14 @@ pub struct UserMode<'a> {
     context: UserContext,
 }
 
-// An instance of `UserMode` is bound to the current task. So it cannot be
+// An instance of `UserMode` is bound to the current task. So it cannot be [`Send`].
 impl<'a> !Send for UserMode<'a> {}
 
 impl<'a> UserMode<'a> {
-    /// Creates a new `UserMode`.
-    pub fn new(user_space: &'a Arc<UserSpace>) -> Self {
+    /// Creates a new `UserMode` on the current task.
+    pub fn new(current_task: &Arc<Task>, user_space: &'a Arc<UserSpace>) -> Self {
```

Comment:
I don't think the constructor should take a caller-given `current_task`. We cannot enforce at the compile time that `current_task` is indeed the current task. And `UserMode::new` is rarely called. So calling `Task::current()` in its constructor is preferable.